weaviate/logger.py

```diff
@@ -1,5 +1,250 @@
+        logger.debug("Headers:")
+        masked_req_headers = _mask_sensitive_headers(dict(request.headers))
+        for key, value in masked_req_headers.items():
+            header_line = f"  {key}: {value}"
+            print(f"DEBUG: About to log header: {header_line}")
+            logger.debug(header_line)
+        
+        # Log request body if present
+        if request.content:
+            print("DEBUG: About to log request body")
+            logger.debug("Body:")
+            try:
+                request_body = json.loads(request.content)
+                body_str = textwrap.indent(json.dumps(request_body, indent=2), "  ")
+                print(f"DEBUG: About to log body: {body_str}")
+                logger.debug(body_str)
+            except json.JSONDecodeError:
+                body_str = textwrap.indent(request.content.decode(errors="ignore"), "  ")
+                print(f"DEBUG: About to log raw body: {body_str}")
+                logger.debug(body_str)
+        
+        # Log response details with clear separation
+        separator = "==================== HTTP Response ==================="
+        print(f"DEBUG: About to log response separator: {separator}")
+        logger.debug(separator)
+        
+        response_line = f"Response: status_code={response.status_code}"
+        print(f"DEBUG: About to log response line: {response_line}")
+        logger.debug(response_line)
+        
+        print("DEBUG: About to log response headers")
+        logger.debug("Headers:")
+        masked_resp_headers = _mask_sensitive_headers(dict(response.headers), is_response=True)
+        for key, value in masked_resp_headers.items():
+            header_line = f"  {key}: {value}"
+            print(f"DEBUG: About to log response header: {header_line}")
+            logger.debug(header_line)
+        
+        separator = "===================================================="
+        print(f"DEBUG: About to log final separator: {separator}")
+        logger.debug(separator)
+        
+        print("DEBUG: Finished logging HTTP event")
+    except Exception as e:
+        print(f"DEBUG: Error in log_http_event: {str(e)}")
+        import traceback
+        print(f"DEBUG: Traceback: {traceback.format_exc()}")
+        logger.debug(f"Error logging HTTP event: {str(e)}")
+
+def log_grpc_event(method_name: str, request: Any, response: Any) -> None:
+    """Log gRPC request and response details using the default logger.
+    
+    Args:
+        method_name: The name of the gRPC method being called
+        request: The gRPC request object
+        response: The gRPC response object (may be None for request-only logging)
+    """
+    print(f"\nDEBUG: log_grpc_event called for method {method_name}")
+    print(f"DEBUG: Logger level: {logger.getEffectiveLevel()}, handlers: {logger.handlers}")
+    print(f"DEBUG: Request type: {type(request)}")
+    print(f"DEBUG: Response type: {type(response)}")
```

Comment:
These are `print` statements and not `logger.debug` and so will show up regardless of the `WEAVIATE_LOG_LEVEL` specification making the client noisy